magda-registry-api/src/test/scala/au/csiro/data61/magda/registry/WebHookProcessingSpec.scala

```diff
@@ -81,27 +87,39 @@ class WebHookProcessorSpec extends ApiSpec {
         status shouldEqual StatusCodes.OK
       }
 
-      val result2 = Await.result(processor.sendSomeNotificationsForOneWebHook("test"), 5 seconds)
-      result2.statusCode should be(Some(StatusCodes.OK))
+      Util.waitUntilDone(actor, "test")
 
       payloads.length shouldBe 1
       payloads(0).events.get.length shouldBe 1
-      payloads(0).records.get.length shouldBe 0
+      payloads(0).records.get.length shouldBe 1
       payloads(0).aspectDefinitions.get.length shouldBe 0
     }
   }
 
-  it("does not duplicate records or aspect definitions") { param =>
-    testWebHook(param, None) { payloads =>
-      val a = AspectDefinition("A", "A", Some(JsObject()))
+  it("does not duplicate records") { param =>
```

Comment:
The meaning of this test changed, too.  Previously it:
1. Did some things that generated multiple events impacting a single record.
2. Ran the webhook processing logic.
3. Asserted that the single payload contained the record only once, rather than (for example) once per event.

But now it expects multiple payloads and combines them all together and just asserts that there aren't any extraneous records listed in any of the payloads (which is pretty unlikely).

I believe I added this test based on an actual bug, so it's worth preserving. Maybe that can be done by setting the webhook inactive while generating the multiple events, and then setting it active again afterward?